src/components/VocabularyTermsList.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  FixedSizeList as List,
+  areEqual,
+  ListChildComponentProps,
+} from "react-window";
+import RouteLink from "./RouteLink";
+import React, { memo } from "react";
+import { DetailItemWrapper } from "./Hierarchy";
+import { Box } from "@material-ui/core";
+import AutoSizer from "react-virtualized-auto-sizer";
+import { makeStyles } from "@material-ui/core/styles";
+import memoize from "memoize-one";
+
+const useStyles = makeStyles((theme) => ({
+  wrapper: {
+    borderColor: "#e0e0e0 !important",
+    "&:hover": {
+      backgroundColor: "#e0e0e0",
+    },
+  },
+}));
+
+const Row = memo(({ data, index, style }: ListChildComponentProps) => {
+  const classes = useStyles();
+  const { items } = data;
+  const item = items[index];
+
+  return (
+    <div style={style}>
+      <Box borderBottom={2} className={classes.wrapper}>
+        <RouteLink
+          to={item.route}
+          variant="h6"
+          style={{
+            display: "block",
+            whiteSpace: "nowrap",
+            overflow: "hidden",
+            textOverflow: "ellipsis",
+          }}
+        >
+          {item.label.cs}
+        </RouteLink>
+      </Box>
+    </div>
+  );
+}, areEqual);
+
+const createItemData = memoize((items) => ({
+  items,
+}));
+
+interface VocabularyTermsListProps {
+  terms: {
+    uri: string;
+    vocabulary: string;
+    label: { cs?: string };
+    route: string;
+  }[];
+}
+
+const VocabularyTermsList: React.FC<VocabularyTermsListProps> = (props) => {
+  const itemData = createItemData(props.terms);
```

Comment:
What exactly is this `memoize` supposed to do here? If it's just converting an array to an object, that's not a heavy operation at all. I'd either just do that, or used `useMemo` hook from React instead of yet another package.

Reply:
I guess you need to pass the same instance of the object to the List perhaps? Still I'd use the `useMemo` hook.